### PR TITLE
feat: Add new method "takePicture"  to Mobile Scanner Controller

### PR DESCRIPTION
**Pull request by Kaspear**
This pull request introduces a new feature that enables the capability to capture pictures using the camera functionality

### CODE REVIEW

File: android/src/main/kotlin/dev/steenbakker/mobile_scanner/MobileScannerHandler.kt [hunk 1]
```diff
@@ -114,13 +119,15 @@ class MobileScannerHandler(
                         }
                     }
                 })
+
```
Comment (wolnikowa123) [inline android/src/main/kotlin/dev/steenbakker/mobile_scanner/MobileScannerHandler.kt:122]: no need for whitespace

File: android/src/main/kotlin/dev/steenbakker/mobile_scanner/MobileScannerHandler.kt [hunk 2]
```diff
@@ -246,4 +253,13 @@ class MobileScannerHandler(
     private fun updateScanWindow(call: MethodCall) {
         mobileScanner!!.scanWindow = call.argument<List<Float>?>("rect")
     }
+
+    private fun takePicture(result: MethodChannel.Result) {
+        photoResult = result
+        try {
+            mobileScanner!!.takePicture(takePictureCallback)
+        } catch (e: TakePictureError) {
+            result.error("MobileScanner", "Error occurred when taking picture!", null)
```
Comment (wolnikowa123) [inline android/src/main/kotlin/dev/steenbakker/mobile_scanner/MobileScannerHandler.kt:262]: Error occurred when taking the picture!*

File: ios/Classes/SwiftMobileScannerPlugin.swift [hunk 3]
```diff
@@ -261,4 +263,21 @@ public class SwiftMobileScannerPlugin: NSObject, FlutterPlugin {
...
+            try mobileScanner.takePicture { imageData in
+                if let data = imageData {
+                    result(FlutterStandardTypedData(bytes: data))
+                } else {
+                    result(nil)
+                }
+            }
+        } catch MobileScannerError.takePictureError {
+            result(FlutterError(code: "MobileScanner",
+                                message: "Error occurred when taking picture!",
```
Comment (wolnikowa123) [inline ios/Classes/SwiftMobileScannerPlugin.swift:278]: *the

**Review by wolnikowa123 — APPROVED** [review]

**Comment by navaronbracke** [comment]
This is not the purpose of this plugin. For that you have the `camera` package.

**Comment by juliansteenbakker** [comment]
Thank you for your effort, but this is beyond the scope of this package, which is aimed at keeping barcode scanning simple, so i'm closing this PR.

**Pull request closed without merging**